frontend/src/views/ElectedCanteens.vue

```diff
@@ -0,0 +1,150 @@
+        { text: "Publication", value: "publicationStatus", sortable: false },
+      ],
+    }
+  },
+  computed: {
+    offset() {
+      return (this.options.page - 1) * this.limit
+    },
+    departmentsString() {
+      const userDepartments = this.$store.state.loggedUser.departments
+      const departmentsArray = jsonDepartments
+        .filter((x) => userDepartments.indexOf(x.departmentCode) > -1)
+        .map((x) => `${x.departmentName} (${x.departmentCode})`)
+      if (departmentsArray.length < 2) return `le département ${departmentsArray.join("")}`
+      return `les départements ${departmentsArray.slice(0, -1).join(", ")} et ${departmentsArray.slice(-1)}`
+    },
+  },
+  methods: {
+    fetchCurrentPage() {
+      this.loading = true
+      return fetch(`/api/v1/electedCanteens/?${this.getApiQueryParams()}`)
+        .then((response) => {
+          if (response.status < 200 || response.status >= 400) throw new Error(`Error encountered : ${response}`)
+          return response.json()
+        })
+        .then((response) => {
+          this.visibleCanteens = response.results
+          this.canteenCount = response.count
+        })
+        .catch((e) => {
+          this.publishedCanteenCount = 0
+          this.visibleCanteens = []
+          this.$store.dispatch("notifyServerError", e)
+        })
+        .finally(() => (this.loading = false))
+    },
+    getApiQueryParams(includePagination = true) {
+      let apiQueryParams = includePagination ? `limit=${this.limit}&offset=${this.offset}` : ""
+      const orderingItems = this.getOrderingItems()
+      if (orderingItems.length > 0) apiQueryParams += `&ordering=${orderingItems.join(",")}`
+      return apiQueryParams
+    },
+    addWatchers() {
+      this.$watch("options", this.onOptionsChange, { deep: true })
+      this.$watch("$route", this.onRouteChange)
+    },
+    onOptionsChange() {
+      this.$router.push({ query: this.getUrlQueryParams() }).catch(() => {})
+    },
+    onRouteChange() {
+      this.populateParametersFromRoute()
+      this.fetchCurrentPage()
+    },
+    getUrlQueryParams() {
+      let urlQueryParams = { page: this.options.page }
+      const orderingItems = this.getOrderingItems()
+      if (orderingItems.length > 0) urlQueryParams["trier-par"] = orderingItems.join(",")
+      return urlQueryParams
+    },
+    getOrderingItems() {
+      let orderParams = []
+      if (this.options.sortBy && this.options.sortBy.length > 0)
+        for (let i = 0; i < this.options.sortBy.length; i++)
+          orderParams.push(this.options.sortDesc[i] ? `-${this.options.sortBy[i]}` : this.options.sortBy[i])
+      return orderParams
+    },
+    populateParametersFromRoute() {
+      const page = this.$route.query.page ? parseInt(this.$route.query.page) : 1
+      let sortBy = []
+      let sortDesc = []
+
+      this.$route.query["trier-par"] &&
+        this.$route.query["trier-par"].split(",").forEach((element) => {
```

Comment:
I don't think there is ever more than one trier-par, and if there is maybe we should use multiple instances in the URL instead of putting them in one (e.g `?...&trier-par[]=siret&trier-par[]=-nom` rather than trier-par=siret,-nom)

Reply:
Currently we don´t have more than one indeed, but only because the `multi-sort` param of the data table is false (by default). If we decided to turn it on we would be ready to do multiiple sort.

While there is no defined standard for multiple values for the same param, there are a couple of reasons why I slightly prefer the comma-separated approach for this case:
- I find it easier to read (since all order values are directly next to each other)
- Easier to see the order of the order parameters (since `a,b` is not the same as `b,a`)
- Shorter URLs
- None of the order parameters will have a comma in them

In other cases - notably for backend filtering - it could be different, since some frameworks take the approach `param=a,b` as an OR and `param=a&param=b` as an AND. But in this case it's just a front-end URL construction.